src/plugins/CoreLoadPlugin.ts

```diff
@@ -54,11 +61,91 @@ function stripPath(basePath: string, path: string): string {
 	return resolved;
 }
 
+interface CallExpressionWithParent {
+	callExpression: CallExpression;
+	path: any[];
+}
+
+/**
+ * Iterate through a Statement and find all the CallExpressions. The call expression and the
+ * AST path to the expressio is saved.
+ *
+ * @param statement
+ * @return {CallExpressionWithParent[]}
+ */
+function findCallExpressions(statement: Program) {
+	const callExpressions: CallExpressionWithParent[] = [];
+
+	function walker(path: any[], item: any) {
+		if (!item) {
+			return;
+		}
+
+		if (item instanceof Array) {
+			item.forEach(arrayItem => walker([ ...path, item ], arrayItem));
+			return;
+		}
+		else if (item.type === 'CallExpression') {
+			callExpressions.push({
+				callExpression: item,
+				path: path
+			});
+		}
+
+		const next = getNextItem(item);
+		next && walker([ ...path, item ], next);
+	}
+
+	walker([], statement);
+
+	return callExpressions;
+}
+
+/**
+ * An object of chunk names and regular expressions. If the requested resource matches the RegExp, the chunk name
+ * will be used.
+ */
+export interface DojoLoadChunkNames {
+	[key: string]: RegExp;
+}
+
+/**
+ * Options for the DojoLoadPlugin
+ */
+export interface DojoLoadPluginOptions {
+	detectLazyLoads?: boolean;
+	chunkNames?: DojoLoadChunkNames;
+	ignoredModules?: string[];
+	basePath?: string;
+}
+
 /**
  * A webpack plugin that forces webpack to ignore `require` passed as a value, and replaces `@dojo/core/load` with a
  * custom function that maps string module IDs to webpack's numerical module IDs.
  */
 export default class DojoLoadPlugin {
+	detectLazyLoads: boolean;
+	lazyChunkNames: DojoLoadChunkNames;
+	ignoredModules: Set<string>;
+
+	constructor(options: DojoLoadPluginOptions = {}) {
+		const { detectLazyLoads, chunkNames, ignoredModules, basePath = '' } = options;
```

Comment:
If you default the options here you won't need the `||` when initialising the properties.